sympy/integrals/meijerint.py

```diff
@@ -2091,6 +2099,11 @@ def meijerint_inversion(f, x, t):
         shift = Add(*exponentials)
         f = Mul(*newargs)
 
+    if f.is_number:
```

Comment:
```diff
- if f.is_number:
+ if x not in f.free_symbols and shift.is_real:-
```

@jksuom maybe this would be better?  `shift.is_real` is to address your boundedness comment.   With this, we could have:
```
>>> a, b = symbols('a, b', real=True)
>>> meijerint.meijerint_inversion(a*exp(b*s), s, t)
a*DiracDelta(b + t)
```

Reply:
It would be better to add it as one of the convergence conditions that are returned from the transform function if that's possible.

Reply:
Perhaps something like this could be returned when `f` is constant: 

    Piecewise(f*DiracDelta(t_ + shift), shift.is real), (None, True))

I think that `None` would be better than an unevaluated integral as it would show that the transform does not exist.

Reply:
You mean `nan`?

Reply:
I thought that `None` would better describe the nonexistence.

Reply:
But None isn't a symbolic object in SymPy.

Maybe we should create some kind of non-convergent integral object for cases like this (we could also have non-convergent Sum).

Reply:
I investigated a bit further.

1.  I don't think you're allowed to pass `None` as the cond in a Piecewise.  (And of course `z.is_real` is `None` for general symbols `z`.)

2.  The closest I could find to a function form of `.is_real` is `Contains(shift, Reals)`.  But these don't interact well with `Piecewise`:

```
>>> r = Symbol('r', real=True)
>>> z = Symbol('z')

>>> Piecewise((6, Contains(r, Reals)), (7, True))    # here Contains evaluates to True
6
```
But
```
>>> Piecewise((6, Contains(r, Reals, evaluate=False)), (7, True))
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-10-a59a6e101eb5> in <module>()
----> 1 Piecewise((6, Contains(r, Reals, evaluate=False)), (7, True))

/home/cbm/src/sympy.git/sympy/functions/elementary/piecewise.pyc in __new__(cls, *args, **options)
    134 
    135         if options.pop('evaluate', True):
--> 136             r = cls.eval(*newargs)
    137         else:
    138             r = None

/home/cbm/src/sympy.git/sympy/functions/elementary/piecewise.pyc in eval(cls, *_args)
    183                     x = free.pop()
    184                     try:
--> 185                         c = c.as_set().as_relational(x)
    186                     except NotImplementedError:
    187                         pass

AttributeError: 'Contains' object has no attribute 'as_relational'
```
And thus also
```
>>> Piecewise((6, Contains(z, Reals)), (7, True))
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-12-c87274518aa0> in <module>()
----> 1 Piecewise((6, Contains(z, Reals)), (7, True))

/home/cbm/src/sympy.git/sympy/functions/elementary/piecewise.pyc in __new__(cls, *args, **options)
    134 
    135         if options.pop('evaluate', True):
--> 136             r = cls.eval(*newargs)
    137         else:
    138             r = None

/home/cbm/src/sympy.git/sympy/functions/elementary/piecewise.pyc in eval(cls, *_args)
    183                     x = free.pop()
    184                     try:
--> 185                         c = c.as_set().as_relational(x)
    186                     except NotImplementedError:
    187                         pass

AttributeError: 'Contains' object has no attribute 'as_relational'
```

Reply:
Contains should work as a boolean, but regardless, the better way to do what you want is `Eq(im(r), 0)`.

Reply:
I opened https://github.com/sympy/sympy/issues/14977 for `Contains`. It should be easy to fix.

Reply:
thanks!  I'll revise with `Eq(im(shift), 0)`

Reply:
I am not sure that an integral object would be a good solution. It seems that returning `None` in case the inverse does not exist would suffice. [`_inverse_laplace_transform`](https://github.com/sympy/sympy/blob/master/sympy/integrals/transforms.py#L1164) would then do the right thing, raise `IntegralTransformError`. Otherwise we could probably return a Piecewise with one component. Only the first one is inspected by the caller. So, maybe something like this could work.

    if shift.is_real != False:
        return Piecewise((f*DiracDelta(t_ + shift), shift.is real))
    else:
        return None

If `shift.is_real == True`, it seems that Piecewise will automatically simplify.